src/Tools/ExternalAccess/ProjectSystem/ProjectSystemTempPECompiler.cs

```diff
@@ -0,0 +1,29 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Host.Mef;
+using Microsoft.VisualStudio.LanguageServices.ProjectSystem;
+
+namespace Microsoft.CodeAnalysis.ExternalAccess.ProjectSystem
+{
+    [Export(typeof(IProjectSystemTempPECompiler))]
+    [Shared]
+    internal sealed class ProjectSystemTempPECompiler : IProjectSystemTempPECompiler
+    {
+        private readonly ITempPECompiler _implementation;
+
+        [ImportingConstructor]
+        [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
+        public ProjectSystemTempPECompiler(ITempPECompiler implementation)
```

Comment:
So this interface isn't even used for anybody other than the project system. We should just move the TempPE stuff either into this layer _entirely_ or just move it to their repo.

Reply:
This will be trivial to verify and implement correctly as a refactoring after the first move to the External Access pattern is complete.